extensions/qute/deployment/src/test/java17/io/quarkus/qute/deployment/records/TemplateRecordConflictTest.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.qute.deployment.records;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.qute.CheckedTemplate;
+import io.quarkus.qute.TemplateException;
+import io.quarkus.qute.TemplateInstance;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class TemplateRecordConflictTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(monkFoo.class, monk_foo.class)
+                    .addAsResource(new StringAsset("Hello {name}!"), "templates/TemplateRecordConflictTest/monk_foo.txt"))
+            .assertException(t -> {
+                Throwable e = t;
+                TemplateException te = null;
+                while (e != null) {
+                    if (e instanceof TemplateException) {
+                        te = (TemplateException) e;
+                        break;
+                    }
+                    e = e.getCause();
+                }
+                assertNotNull(te);
+                assertTrue(
+                        te.getMessage().contains(
+                                "Multiple checked templates exist for the template path TemplateRecordConflictTest/monk_foo"),
+                        te.getMessage());
```

Comment:
```suggestion
		assertThat(t).hasRootCauseMessage("Multiple checked templates exist for the template path TemplateRecordConflictTest/monk_foo");
```

Reply:
We can't use `hasRootCauseMessage()` because we only match the start of an exception message...